### PR TITLE
Docs: Update `Architecture` section

### LINKED ISSUE #1406: Update `architecture` diagram
Ref: https://github.com/webhintio/hint/blob/da7cb4903ffd9b36c644d87046598604b2555669/packages/hint/docs/contributor-guide/getting-started/images/architecture.svg

Still contains references to `sonarwhal`.

### PR DESCRIPTION
**Pull request by molant**
Fix #1406

## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [x] Added/Updated related documentation.
- ~~Added/Updated related tests.~~

## Short description of the change(s)

Update the architecture section as well as the diagram.
I've optimized the svg but probable @alrra will do some more magic.
I haven't removed the old diagram just in case we want to update it in the future although it's not very easily editable.

This is how it looks like now:

CLI sequence diagram

### CODE REVIEW

**Comment by molant** [comment]
@CKGrafico @isabellachen you might want to give your feedback on this one as some of the most recent contributors 😊

File: packages/hint/docs/contributor-guide/getting-started/architecture.md [hunk 1]
```diff
@@ -1,78 +1,81 @@
...
-   `event`s related to this and the resource loading process.
-1. If a `parser` has subscribed to one of the emitted `event`s, it will parse
-   that resource and `emit` new `event`s.
-1. `hint`s can subscribe to any `event` no matter where it's comming from
-   (`connector` or `parser`). If they find an issue, it will be reported via the
-   `report` method.
-1. Once `webhint` has finished analyzing the website, the results are returned
-   to `CLI` that will print them using the configured `formatter`(s).
+   `executeOn` method.
+1. `engine` then calls the `collect` method (`async`) of the configured
```
Comment (sarvaje) [inline packages/hint/docs/contributor-guide/getting-started/architecture.md]:
```suggestion
1. `Engine` then calls the `collect` method (`async`) of the configured
```

**Comment by CKGrafico** [comment]
I really like it, my unique suggestion could be that instead hint 1, hint 2 use names or almost a comment to see the relation with the connector or the parser.
image

**Comment by CKGrafico** [comment]
@molant in your schema, hint 1 is for connector and hint 2 is for parser (I think) but if I read them as hint 1 and hint 2 I can be a little confused.

Is true that with the arrows is explained, but examples helps (hint for connector event and hint for parser event)

**Comment by molant** [comment]
PR updated with a new diagram. This is how it looks like now. Hope it is clearer now @CKGrafico !

image

**Comment by molant** [comment]
@alrra the diagram is not optimized, I'll wait until everyone is OK with it before optimizing it.

**Comment by alrra** [comment]
>  I'll wait until everyone is OK with it before optimizing it.

Yes, that was my plan.

File: packages/hint/docs/contributor-guide/getting-started/architecture.md [hunk 2]
```diff
@@ -1,78 +1,81 @@
...
-* `connector`: Is the way in which `webhint` obtains information about the
-  DOM, network information, resources, etc. The underlying technique (debugging
-  protocol, web driver, etc.) to access this data does not matter to the rest
-  of the system.
-  [Learn how to develop a `connector`][new connector].
-* `formatter`: Transforms the results into something useful to the user. It
-  could be as simple as printing out the results to the command line, or
-  something more complex like creating an HTML report.
-  [Learn how to developer a `formatter`][new formatter].
+  document, image, request, etc.). E.g.: Verify that the HTML document
```
Comment (alrra) [inline packages/hint/docs/contributor-guide/getting-started/architecture.md]: Add `configuration files` to the list.

File: packages/hint/docs/contributor-guide/getting-started/architecture.md [hunk 3]
```diff
@@ -1,78 +1,81 @@
...
 
-When a user invokes `hint https://example.com` basically what happens is
-this:
+All these pieces communicate with each other via [`event`s][events].
+`Engine`, which extends from `EventEmitter`, is the one enabling this
+interaction. Most of the new work is done in one of the previous
+components as `Engine` should be kept as simple as possible. The
+entities emitting events are `connector`s and `parser`s. The ones that
+subscribe to `event`s are `hint`s and `parser`s.
+
```
Comment (alrra) [inline packages/hint/docs/contributor-guide/getting-started/architecture.md]: Should we add a quick note about `configurations`?

File: packages/hint/docs/contributor-guide/getting-started/architecture.md (hunk 2 above)
Comment (molant) [reply packages/hint/docs/contributor-guide/getting-started/architecture.md]: I wasn't sure about the configuration files as those are just JSON files in the system, ie: we can run webhint without a config file as long as we pass the resources.
I see you have other comments about `confiurations` that I think should be addressed in another PR.

File: packages/hint/docs/contributor-guide/getting-started/architecture.md (hunk 3 above)
Comment (molant) [reply packages/hint/docs/contributor-guide/getting-started/architecture.md]: I talk a bit about `configurations` a bit later. Maybe we should have a `how to create your own configuration` section but I think that should go in another PR.

Comment (alrra) [reply packages/hint/docs/contributor-guide/getting-started/architecture.md]: >  in another PR.

Works for me.

File: packages/hint/docs/contributor-guide/getting-started/architecture.md (hunk 2 above)
Comment (alrra) [reply packages/hint/docs/contributor-guide/getting-started/architecture.md]: Configuration files as in `.<something>rc` files.

Comment (molant) [reply packages/hint/docs/contributor-guide/getting-started/architecture.md]: Ahh, got it.

Comment (molant) [reply packages/hint/docs/contributor-guide/getting-started/architecture.md]: Done

File: packages/hint/docs/contributor-guide/getting-started/architecture.md [hunk 4]
```diff
@@ -1,78 +1,75 @@
...
 
-[default configuration]: ../../user-guide/#default-configuration
-[events]: ./events/
-[new connector]: ../how-to/connector/
-[new formatter]:../how-to/formatter/
-[new parser]: ../how-to/parser/
-[new hint]: ../how-to/hint/
-[npm]: https://www.npmjs.com/
-[typescript]: https://www.typescriptlang.org/
+[default configuration]: ../../user-guide/#default-configuration [events]:
```
Comment (alrra) [inline packages/hint/docs/contributor-guide/getting-started/architecture.md]: This needs to be fixed.

Comment (molant) [reply packages/hint/docs/contributor-guide/getting-started/architecture.md]: Done, I'm using a new extension that automatically wraps content at 80 characters and looks like it messes link labels 😔

**Comment by alrra** [comment]
> optimizing it

Done in https://github.com/webhintio/hint/commit/f6af752583db5744212dd5c438b67fdf0e995b20.

**Pull request closed without merging**